Framework/API/src/ExperimentInfo.cpp

```diff
@@ -187,23 +187,27 @@ void checkDetectorInfoSize(const Instrument &instr,
                              "instrument");
 }
 
-boost::shared_ptr<Beamline::ComponentInfo>
-makeComponentInfo(const Instrument &oldInstr,
-                  const API::DetectorInfo &detectorInfo,
-                  std::vector<Geometry::ComponentID> &componentIds) {
+boost::shared_ptr<Beamline::ComponentInfo> makeComponentInfo(
```

Comment:
Yeah, but I still create `unique_ptr` and then dump it into `shared_ptr`. `DetectorInfo` is currently held as a `shared_ptr` only for legacy workarounds, namely for storing it in `Instrument` in `ExperimentInfo::getInstrument`. Eventually it will not be shared, since it does all sharing internally.